### PR TITLE
replaced Generator by EntityClassGenerator in MakerResetPassword

### PR DESCRIPTION
**Pull request by archeoprog**
Q | A
-- | --
Branch? | master
Bug fix? | no
New feature? | - yes (internal)
Deprecations? | - no
Tickets | - no
License | MIT
Doc PR | - no need

At the request of Ryan Weaver.

I replaced the use of `Generator` by `EntityClassGenerator`. 

We then needed to use `ClassSourceManipulator` and i propose to add 3 methods to this class, tests included.

1) `$manipulator->clearClassNodeStmts()`
Very short method to remove all `Node\Stmts` in the class and update the source code.

2) `$manipulator->addTrait()`
Add a `use Trait;` at first in the class.

3) `$manipulator->addConstructor()`
Is a shortcut to add a constructor in the right place.

I also propose to add two optional parameters to the `$manipulator->addMethodBuilder()` method: `array $params = [], string $methodBody = null;`

Allowing to build methods more directly when needed.

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# In all environments, the following files are loaded if they exist,
```
Comment (jrushlow) [inline .env:1]: I think you may have accidentally committed this `.env` file.

**Comment by jrushlow** [comment]
PHP-CS-Fixer is failing in CI, perhaps run `vendor/bin/php-cs-fixer fix --diff --diff-format=udiff --dry-run` locally to inspect the proposed changes, then run that command again without `--dry-run`

File: composer.json [hunk 2]
```diff
@@ -30,7 +32,7 @@
         "friendsofphp/php-cs-fixer": "^2.8",
         "friendsoftwig/twigcs": "^3.1.2",
         "symfony/http-client": "^4.3|^5.0",
-        "symfony/phpunit-bridge": "^4.3|^5.0",
+        "symfony/phpunit-bridge": "^5.1@dev",
```
Comment (jrushlow) [inline composer.json:35]: Why the change in version constraints? We may want to leave this at `^4.3|^5.0`. `minimum-stability: dev` will use the latest dev releases.

File: composer.json [hunk 3]
```diff
@@ -14,13 +14,15 @@
     "minimum-stability": "dev",
     "require": {
         "php": "^7.0.8",
+        "doctrine/annotations": "^1.11@dev",
```
Comment (jrushlow) [inline composer.json:17]: I think this may have been added by accident. But I could be mistaken. Either way, requiring `@dev` would be wise here, as app in production would then be forced to use unstable branches.  Same for `symfony/flex`

Comment (archeoprog) [reply composer.json:17]: Yes, sorry. i am looking for fixing all theses errors, it is my first contribution to a collectif project i am a bit lost...

Comment (jrushlow) [reply composer.json:17]: hey no problem, if you have any questions feel free to DM on slack `jrushlow`

**Comment by archeoprog** [comment]
First experience

**Comment by archeoprog** [comment]
Mishandling, major conflicts.
